res/controllers/Gemini-G4V-scripts.js

```diff
@@ -0,0 +1,562 @@
+            case 0x1F:
+                // Not mapped, for future use
+                // TODO - Four buttons to enable effect racks
+                break;
+            case 0x20:
+                // Not mapped, for future use
+                break;
+            case 0x21:
+                this.deck.padMode(g4v.Deck.prototype.modes.beatJump);
+                break;
+            case 0x22:
+                this.deck.padMode(g4v.Deck.prototype.modes.manualLoop);
+                break;
+            }
+        },
+    });
+    // Deck togle button (not connected to CO)
+    this.deckToggleBtn = new components.Component({
+        deck: this,
+        input: function(_channel, _control, value, _status, _group) {
+            if (!value) { return; }
+            this.deck.toggle();
+            const index = this.deck.deckNumbers.indexOf(parseInt(
+                script.channelRegEx.exec(this.deck.currentDeck)[1]));
+            if (index === 1) { midi.sendShortMsg(0x8F + midiChannel, 0x26, 0x7F); } else { midi.sendShortMsg(0x8F + midiChannel, 0x26, 0x00); }
+        },
+    });
+
+    this.jogTopBtn = new components.Component({
+        input: function(_channel, _control, value, _status, _group) {
+            switch (value) {
+            case 0x7f: {
+                const intervalsPerRev = 400;
+                const rpm = 30+1/3;
+                const alpha = (1.0/8);
+                const beta = (alpha / 32);
+                engine.scratchEnable(script.deckFromGroup(this.group), intervalsPerRev, rpm, alpha, beta);
+                break;
+            }
+            case 0x00:
+                engine.scratchDisable(script.deckFromGroup(this.group), 1);
+                break;
+            }
+        },
+    });
+    this.jogEnc = new components.Encoder({
+        midi: [0x89 + midiChannel, 0x06],
+        factor: 1.4,
+        input: function(_channel, _control, value, _status, _group) {
+            if (engine.isScratching(script.deckFromGroup(this.group))) {
+                engine.scratchTick(script.deckFromGroup(this.group), (value-64)*this.factor);
+            } else {
+                engine.setValue(this.group, "jog", (value === 0x41 ? -1 : 1)*0.2);
+            }
+        },
+        shift: function() { this.factor = 10; },
+        unshift: function() { this.factor = 1.4; },
+    });
+
+    // Changes pad layer
+    this.padMode = function(mode) {
+        // Switch off all the mode leds
+        midi.sendShortMsg(0x8F + midiChannel, 0x1B, 0x00);
+        midi.sendShortMsg(0x8F + midiChannel, 0x1C, 0x00);
+        midi.sendShortMsg(0x8F + midiChannel, 0x1D, 0x00);
+        midi.sendShortMsg(0x8F + midiChannel, 0x1E, 0x00);
+
+        // Switch off all pad leds
+        if (this.pads !== undefined) {
+            this.pads.shutdown();
+        }
+
+        switch (mode) {
+        case g4v.Deck.prototype.modes.cue:
+            this.pads = this.padsCue;
+            midi.sendShortMsg(0x8F + midiChannel, 0x1B, 0x7F);
```

Comment:
Instead of managing the LED here, put this into the constructor of each padmode (then overwrite the shutdown function to switch of that LED again). This follows separation of concerns better (and follows the modern RAII pattern (which is kinda overkill here but still good practice I'd say)).